.github/workflows/selfhosted.yml

```diff
@@ -54,9 +54,30 @@ jobs:
         conda create --prefix=testenv python=${{ matrix.python-version }}
         conda activate ./testenv
         python -m pip install --upgrade pip
-        pip install pytest
+        pip install pytest-cov
         pip install git+https://github.com/qiboteam/qibo
         pip install .[tiiq]
     - name: Test ${{ matrix.platform }}
       run: |
-        srun -p ${{ matrix.platform }} pytest src/qibolab --pyargs qibolab -m qpu --platform ${{ matrix.platform }} -rx
+        # create script for slurm
+        platform=${{ matrix.platform }}
+        echo \#\!/bin/bash >> ci_tests
+        echo >> ci_tests
+        echo "mkdir /tmp/qibolab" >> ci_tests
+        echo "ln -sr src/qibolab/tests /tmp/qibolab" >> ci_tests
+        echo "cd /tmp/qibolab" >> ci_tests
+        echo "pytest --cov=qibolab --cov-report=xml --platform $platform -rx" >> ci_tests
+        echo "cd -" >> ci_tests
+        echo "mv /tmp/qibolab/coverage.xml ." >> ci_tests
+        echo "rm -r /tmp/qibolab/" >> ci_tests
+        # execute script in the queue
+        chmod +x ci_tests
```

Comment:
Since you're just replacing `$platform`, the script is better committed to the repo, and then you can `sed` replace the placeholder (maybe call it `$PLATFORM` to make it more evident).

Reply:
Git preserves the file mode, so even the `chmod +x` can be commited.